Test failures in VarHandle-related sanity.openjdk components

A number of `sanity.openjdk` tests have begun failing in the JITServer nightly tests. I believe this started last night (so the May 22 tests). This is on all architectures.

One of the failed runs is https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal/968/, with failed components:

- `jdk_lang_VarHandleTest_j9_1`
- `jdk_lang_1`
- `jdk_lang_VarHandleTest_j9_0`
- `jdk_lang_0`
- `jdk_lang_j9_0`

I think these are all in VarHandle-related subtests. (There was an additional failure in `jdk_util_0`, but it was a timeout in `CountLargeTest` that I've seen before).

Console log for one of the failures in `jdk_lang_1`, for example:

```
03:56:11  test VarHandleTestMethodHandleAccessLong.testAccess("VarHandle -> VarHandle.toMethodHandle -> Static field", VarHandleBaseTest$MethodHandleAccessTestCase@77718d93): failure
03:56:11  java.lang.AssertionError: getAndSet long expected [81985529216486895] but found [0]
03:56:11  	at org.testng.Assert.fail(Assert.java:99)
03:56:11  	at org.testng.Assert.failNotEquals(Assert.java:1037)
03:56:11  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
03:56:11  	at org.testng.Assert.assertEquals(Assert.java:122)
03:56:11  	at org.testng.Assert.assertEquals(Assert.java:797)
03:56:11  	at VarHandleTestMethodHandleAccessLong.testStaticField(VarHandleTestMethodHandleAccessLong.java:587)
03:56:11  	at VarHandleBaseTest$AccessTestCase.testAccess(VarHandleBaseTest.java:420)
03:56:11  	at VarHandleTestMethodHandleAccessLong.testAccess(VarHandleTestMethodHandleAccessLong.java:119)
03:56:11  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
03:56:11  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
03:56:11  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
03:56:11  	at java.base/java.lang.reflect.Method.invoke(Method.java:574)
03:56:11  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
03:56:11  	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:599)
03:56:11  	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
03:56:11  	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
03:56:11  	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
03:56:11  	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
03:56:11  	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
03:56:11  	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
03:56:11  	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
03:56:11  	at org.testng.TestRunner.privateRun(TestRunner.java:764)
03:56:11  	at org.testng.TestRunner.run(TestRunner.java:585)
03:56:11  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
03:56:11  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
03:56:11  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
03:56:11  	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
03:56:11  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
03:56:11  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
03:56:11  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
03:56:11  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
03:56:11  	at org.testng.TestNG.runSuites(TestNG.java:1069)
03:56:11  	at org.testng.TestNG.run(TestNG.java:1037)
03:56:11  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:102)
03:56:11  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:58)
03:56:11  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
03:56:11  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
03:56:11  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
03:56:11  	at java.base/java.lang.reflect.Method.invoke(Method.java:574)
03:56:11  	at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
03:56:11  	at java.base/java.lang.Thread.run(Thread.java:857)
03:56:11  STDERR:
03:56:11  Unhandled exception
03:56:11  Type=Segmentation error vmState=0x00040000
03:56:11  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000080
03:56:11  Handler1=00007F89DA405CB0 Handler2=00007F89DA35A740 InaccessibleAddress=0000000000000000
03:56:11  RDI=00007F89D424B9A0 RSI=00007F89DA323120 RAX=000800007F893818 RBX=0000000000000000
03:56:11  RCX=00007F89D44B1BCA RDX=0000000000000008 R8=00007F89380D1200 R9=00007F89D3D0CB40
03:56:11  R10=000000000000002A R11=00007F89B407B998 R12=00007F89D4398200 R13=00007F89D44B4E00
03:56:11  R14=00007F89D3D0D438 R15=00007F89990FE0E0
03:56:11  RIP=00007F89DA48A10C GS=0000 FS=0000 RSP=00007F89990FDD60
03:56:11  EFlags=0000000000010206 CS=0033 RBP=00007F89B4209AD8 ERR=0000000000000000
03:56:11  TRAPNO=000000000000000D OLDMASK=0000000000000000 CR2=0000000000000000
03:56:11  xmm0=00007f89381a0e73 (f: 941231744.000000, d: 6.928151e-310)
03:56:11  xmm1=00007f89381b9b68 (f: 941333376.000000, d: 6.928151e-310)
03:56:11  xmm2=00007f89da5f7f30 (f: 3663691520.000000, d: 6.928285e-310)
03:56:11  xmm3=00007f8938215668 (f: 941708928.000000, d: 6.928151e-310)
03:56:11  xmm4=00007f89d424d200 (f: 3559182848.000000, d: 6.928280e-310)
03:56:11  xmm5=00007f8938009ba0 (f: 939563904.000000, d: 6.928150e-310)
03:56:11  xmm6=00007f89381b9b68 (f: 941333376.000000, d: 6.928151e-310)
03:56:11  xmm7=00007f8938009b88 (f: 939563904.000000, d: 6.928150e-310)
03:56:11  xmm8=616a2e6e75732e6d (f: 1970482816.000000, d: 1.840433e+161)
03:56:11  xmm9=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm10=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm11=ca62c1d6ca62c1d6 (f: 3395469824.000000, d: -2.193092e+50)
03:56:11  xmm12=0000013d00000140 (f: 320.000000, d: 6.726727e-312)
03:56:11  xmm13=000001380000013f (f: 319.000000, d: 6.620627e-312)
03:56:11  xmm14=0000000008001800 (f: 134223872.000000, d: 6.631540e-316)
03:56:11  xmm15=000001420000013b (f: 315.000000, d: 6.832826e-312)
03:56:11  Module=/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9vm29.so
03:56:11  Module_base_address=00007F89DA3C2000
03:56:11  Target=2_90_20240521_1029 (Linux 5.4.0-181-generic)
03:56:11  CPU=amd64 (4 logical CPUs) (0x1f096c000 RAM)
03:56:11  ----------- Stack Backtrace -----------
03:56:11  _ZN26VM_BytecodeInterpreterFull3runEP10J9VMThread+0x3dbc (0x00007F89DA48A10C [libj9vm29.so+0xc810c])
03:56:11  bytecodeLoopFull+0xca (0x00007F89DA48633A [libj9vm29.so+0xc433a])
03:56:11   (0x00007F89DA55B2A2 [libj9vm29.so+0x1992a2])
03:56:11  ---------------------------------------
03:56:11  JVMDUMP039I Processing dump event "gpf", detail "" at 2024/05/22 00:56:01 - please wait.
03:56:11  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/core.20240522.005601.1621137.0001.dmp' in response to an event
03:56:11  #0: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x92db45) [0x7f89d952db45]
03:56:11  #1: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x939ef0) [0x7f89d9539ef0]
03:56:11  #2: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x1623d9) [0x7f89d8d623d9]
03:56:11  #3: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9prt29.so(+0x29978) [0x7f89da35a978]
03:56:11  #4: /lib/x86_64-linux-gnu/libpthread.so.0(+0x14420) [0x7f89dac6b420]
03:56:11  #5: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x247686) [0x7f89d8e47686]
03:56:11  #6: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x24948a) [0x7f89d8e4948a]
03:56:11  #7: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8cf6a4) [0x7f89d94cf6a4]
03:56:11  #8: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8ec0f6) [0x7f89d94ec0f6]
03:56:11  #9: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x92019a) [0x7f89d952019a]
03:56:11  #10: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8f40df) [0x7f89d94f40df]
03:56:11  #11: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8cb16b) [0x7f89d94cb16b]
03:56:11  #12: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8e7e27) [0x7f89d94e7e27]
03:56:11  #13: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x92019a) [0x7f89d952019a]
03:56:11  #14: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x920675) [0x7f89d9520675]
03:56:11  #15: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x920d6e) [0x7f89d9520d6e]
03:56:11  #16: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x921210) [0x7f89d9521210]
03:56:11  #17: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x7abd75) [0x7f89d93abd75]
03:56:11  #18: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x7ac13f) [0x7f89d93ac13f]
03:56:11  #19: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x7ada93) [0x7f89d93ada93]
03:56:11  #20: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x59f865) [0x7f89d919f865]
03:56:11  #21: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x17861f) [0x7f89d8d7861f]
03:56:11  #22: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x179651) [0x7f89d8d79651]
03:56:11  #23: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9prt29.so(+0x2a3c9) [0x7f89da35b3c9]
03:56:11  #24: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x177192) [0x7f89d8d77192]
03:56:11  #25: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x1774d8) [0x7f89d8d774d8]
03:56:11  #26: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x1763fb) [0x7f89d8d763fb]
03:56:11  #27: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x176762) [0x7f89d8d76762]
03:56:11  #28: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x176812) [0x7f89d8d76812]
03:56:11  #29: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9prt29.so(+0x2a3c9) [0x7f89da35b3c9]
03:56:11  Unhandled exception
03:56:11  Type=Segmentation error vmState=0x000509ff
03:56:11  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000080
03:56:11  Handler1=00007F89DA405CB0 Handler2=00007F89DA35A740 InaccessibleAddress=0000000000000000
03:56:11  RDI=00007F89D407CD34 RSI=00007F89D44B1BCA RAX=00007F89D44B1BCA RBX=00007F89D811DDE8
03:56:11  RCX=00007F89D8124780 RDX=000800007F893818 R8=00007F89D8340A20 R9=0000000000000000
03:56:11  R10=0000000000000000 R11=0000000000000040 R12=00007F89D407CD34 R13=00007F89D8124780
03:56:11  R14=00007F8932721910 R15=0000000000000006
03:56:11  RIP=00007F89D8E47686 GS=0000 FS=0000 RSP=00007F89D811DDB0
03:56:11  EFlags=0000000000010202 CS=0033 RBP=00007F89D4428760 ERR=0000000000000000
03:56:11  TRAPNO=000000000000000D OLDMASK=0000000000000000 CR2=0000000000000000
03:56:11  xmm0=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm1=0000000000000008 (f: 8.000000, d: 3.952525e-323)
03:56:11  xmm2=ffffffffffffffff (f: 4294967296.000000, d: -nan)
03:56:11  xmm3=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm4=00007f8932680c30 (f: 845679680.000000, d: 6.928146e-310)
03:56:11  xmm5=00007f89326f14c0 (f: 846140608.000000, d: 6.928146e-310)
03:56:11  xmm6=00007f89d811e3a0 (f: 3625051136.000000, d: 6.928283e-310)
03:56:11  xmm7=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm8=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm9=6c6c6c6c6c6c3645 (f: 1819031168.000000, d: 1.913749e+214)
03:56:11  xmm10=0e23430b020d0043 (f: 34406468.000000, d: 1.444350e-240)
03:56:11  xmm11=0000015200000151 (f: 337.000000, d: 7.172346e-312)
03:56:11  xmm12=0000013d00000140 (f: 320.000000, d: 6.726727e-312)
03:56:11  xmm13=000001380000013f (f: 319.000000, d: 6.620627e-312)
03:56:11  xmm14=0000000008001800 (f: 134223872.000000, d: 6.631540e-316)
03:56:11  xmm15=000001420000013b (f: 315.000000, d: 6.832826e-312)
03:56:11  Module=/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so
03:56:11  Module_base_address=00007F89D8C00000
03:56:11  
03:56:11  Method_being_compiled=java/lang/invoke/LambdaForm$MH/0x000000003834e680.invoke(Ljava/lang/Object;J)J
03:56:11  Target=2_90_20240521_1029 (Linux 5.4.0-181-generic)
03:56:11  CPU=amd64 (4 logical CPUs) (0x1f096c000 RAM)
03:56:11  ----------- Stack Backtrace -----------
03:56:11  #0: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x92db45) [0x7f89d952db45]
03:56:11  #1: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x939ef0) [0x7f89d9539ef0]
03:56:11  #2: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x1623d9) [0x7f89d8d623d9]
03:56:11  #3: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9prt29.so(+0x29978) [0x7f89da35a978]
03:56:11  #4: /lib/x86_64-linux-gnu/libpthread.so.0(+0x14420) [0x7f89dac6b420]
03:56:11  #5: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x247686) [0x7f89d8e47686]
03:56:11  #6: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x24948a) [0x7f89d8e4948a]
03:56:11  #7: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8cf6a4) [0x7f89d94cf6a4]
03:56:11  #8: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8ec0f6) [0x7f89d94ec0f6]
03:56:11  #9: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x92019a) [0x7f89d952019a]
03:56:11  #10: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8f40df) [0x7f89d94f40df]
03:56:11  #11: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8cb16b) [0x7f89d94cb16b]
03:56:11  #12: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x8e7e27) [0x7f89d94e7e27]
03:56:11  #13: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x92019a) [0x7f89d952019a]
03:56:11  #14: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x920675) [0x7f89d9520675]
03:56:11  #15: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x920d6e) [0x7f89d9520d6e]
03:56:11  #16: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x921210) [0x7f89d9521210]
03:56:11  #17: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x7abd75) [0x7f89d93abd75]
03:56:11  #18: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x7ac13f) [0x7f89d93ac13f]
03:56:11  #19: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x7ada93) [0x7f89d93ada93]
03:56:11  #20: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x59f865) [0x7f89d919f865]
03:56:11  #21: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x17861f) [0x7f89d8d7861f]
03:56:11  #22: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x179651) [0x7f89d8d79651]
03:56:11  #23: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9prt29.so(+0x2a3c9) [0x7f89da35b3c9]
03:56:11  #24: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x177192) [0x7f89d8d77192]
03:56:11  #25: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x1774d8) [0x7f89d8d774d8]
03:56:11  #26: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x1763fb) [0x7f89d8d763fb]
03:56:11  #27: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x176762) [0x7f89d8d76762]
03:56:11  #28: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so(+0x176812) [0x7f89d8d76812]
03:56:11  #29: /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9prt29.so(+0x2a3c9) [0x7f89da35b3c9]
03:56:11  Unhandled exception
03:56:11  Type=Segmentation error vmState=0x000509ff
03:56:11  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000080
03:56:11  Handler1=00007F89DA405CB0 Handler2=00007F89DA35A740 InaccessibleAddress=0000000000000000
03:56:11  RDI=00007F89D407CD34 RSI=00007F89D44B1BCA RAX=00007F89D44B1BCA RBX=00007F89D831FDE8
03:56:11  RCX=00007F89D8326780 RDX=000800007F893818 R8=00007F89D8340140 R9=0000000000000000
03:56:11  R10=0000000000000000 R11=0000000000000040 R12=00007F89D407CD34 R13=00007F89D8326780
03:56:11  R14=00007F89997EB530 R15=0000000000000006
03:56:11  RIP=00007F89D8E47686 GS=0000 FS=0000 RSP=00007F89D831FDB0
03:56:11  EFlags=0000000000010202 CS=0033 RBP=00007F899400BBB0 ERR=0000000000000000
03:56:11  TRAPNO=000000000000000D OLDMASK=0000000000000000 CR2=0000000000000000
03:56:11  xmm0=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm1=0000000000000008 (f: 8.000000, d: 3.952525e-323)
03:56:11  xmm2=ffffffffffffffff (f: 4294967296.000000, d: -nan)
03:56:11  xmm3=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm4=00007f899977bc30 (f: 2574760960.000000, d: 6.928231e-310)
03:56:11  xmm5=00007f89997fc4c0 (f: 2575287552.000000, d: 6.928231e-310)
03:56:11  xmm6=00007f89d83203a0 (f: 3627156480.000000, d: 6.928283e-310)
03:56:11  xmm7=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm8=6e614e656e696c6e (f: 1852402816.000000, d: 5.004600e+223)
03:56:11  xmm9=0000000000000000 (f: 0.000000, d: 0.000000e+00)
03:56:11  xmm10=6c6c6c6c6c6c6c6c (f: 1819044992.000000, d: 1.913749e+214)
03:56:11  xmm11=0000015200000151 (f: 337.000000, d: 7.172346e-312)
03:56:11  xmm12=0000013d00000140 (f: 320.000000, d: 6.726727e-312)
03:56:11  xmm13=000001380000013f (f: 319.000000, d: 6.620627e-312)
03:56:11  xmm14=0000000008001800 (f: 134223872.000000, d: 6.631540e-316)
03:56:11  xmm15=000001420000013b (f: 315.000000, d: 6.832826e-312)
03:56:11  Module=/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/lib/default/libj9jit29.so
03:56:11  Module_base_address=00007F89D8C00000
03:56:11  
03:56:11  Method_being_compiled=java/lang/invoke/LambdaForm$MH/0x0000000038348bd0.invoke(Ljava/lang/Object;J)V
03:56:11  Target=2_90_20240521_1029 (Linux 5.4.0-181-generic)
03:56:11  CPU=amd64 (4 logical CPUs) (0x1f096c000 RAM)
03:56:11  ----------- Stack Backtrace -----------
03:56:11  _ZN7TR_J9VM21getBaseComponentClassEP19TR_OpaqueClassBlockRi+0x26 (0x00007F89D8E47686 [libj9jit29.so+0x247686])
03:56:11  _ZN11TR_J9VMBase17getClassSignatureEP19TR_OpaqueClassBlockP9TR_Memory+0x2a (0x00007F89D8E4948A [libj9jit29.so+0x24948a])
03:56:11  _ZL25addKnownObjectConstraintsPN3OMR16ValuePropagationEPN2TR4NodeE+0x2e4 (0x00007F89D94CF6A4 [libj9jit29.so+0x8cf6a4])
03:56:11  _Z15constrainIaloadPN3OMR16ValuePropagationEPN2TR4NodeE+0x5a6 (0x00007F89D94EC0F6 [libj9jit29.so+0x8ec0f6])
03:56:11  _ZN3OMR16ValuePropagation10launchNodeEPN2TR4NodeES3_i+0x9a (0x00007F89D952019A [libj9jit29.so+0x92019a])
03:56:11  _Z17constrainChildrenPN3OMR16ValuePropagationEPN2TR4NodeE+0x3f (0x00007F89D94F40DF [libj9jit29.so+0x8f40df])
03:56:11  _ZL15handleNullCheckPN3OMR16ValuePropagationEPN2TR4NodeEb+0xdb (0x00007F89D94CB16B [libj9jit29.so+0x8cb16b])
03:56:11  _Z16constrainNullChkPN3OMR16ValuePropagationEPN2TR4NodeE+0x17 (0x00007F89D94E7E27 [libj9jit29.so+0x8e7e27])
03:56:11  _ZN3OMR16ValuePropagation10launchNodeEPN2TR4NodeES3_i+0x9a (0x00007F89D952019A [libj9jit29.so+0x92019a])
03:56:11  _ZN3OMR16ValuePropagation12processTreesEPN2TR7TreeTopES3_+0x1a5 (0x00007F89D9520675 [libj9jit29.so+0x920675])
03:56:11  _ZN2TR21LocalValuePropagation12processBlockEPNS_7TreeTopE+0x2fe (0x00007F89D9520D6E [libj9jit29.so+0x920d6e])
03:56:11  _ZN2TR21LocalValuePropagation7performEv+0x90 (0x00007F89D9521210 [libj9jit29.so+0x921210])
03:56:11  _ZN3OMR9Optimizer19performOptimizationEPK20OptimizationStrategyiii.localalias+0x855 (0x00007F89D93ABD75 [libj9jit29.so+0x7abd75])
03:56:11  _ZN3OMR9Optimizer19performOptimizationEPK20OptimizationStrategyiii.localalias+0xc1f (0x00007F89D93AC13F [libj9jit29.so+0x7ac13f])
03:56:11  _ZN3OMR9Optimizer8optimizeEv+0x1b3 (0x00007F89D93ADA93 [libj9jit29.so+0x7ada93])
03:56:11  _ZN3OMR11Compilation7compileEv+0xa25 (0x00007F89D919F865 [libj9jit29.so+0x59f865])
03:56:11  _ZN2TR28CompilationInfoPerThreadBase7compileEP10J9VMThreadPNS_11CompilationEP17TR_ResolvedMethodR11TR_J9VMBaseP19TR_OptimizationPlanRKNS_16SegmentAllocatorE+0x4bf (0x00007F89D8D7861F [libj9jit29.so+0x17861f])
03:56:11  _ZN2TR28CompilationInfoPerThreadBase14wrappedCompileEP13J9PortLibraryPv+0x381 (0x00007F89D8D79651 [libj9jit29.so+0x179651])
03:56:11  omrsig_protect+0x239 (0x00007F89DA35B3C9 [libj9prt29.so+0x2a3c9])
03:56:11  _ZN2TR28CompilationInfoPerThreadBase7compileEP10J9VMThreadP21TR_MethodToBeCompiledRN2J917J9SegmentProviderE+0x372 (0x00007F89D8D77192 [libj9jit29.so+0x177192])
03:56:11  _ZN2TR24CompilationInfoPerThread12processEntryER21TR_MethodToBeCompiledRN2J917J9SegmentProviderE+0x128 (0x00007F89D8D774D8 [libj9jit29.so+0x1774d8])
03:56:11  _ZN2TR24CompilationInfoPerThread14processEntriesEv+0x35b (0x00007F89D8D763FB [libj9jit29.so+0x1763fb])
03:56:11  _ZN2TR24CompilationInfoPerThread3runEv+0x42 (0x00007F89D8D76762 [libj9jit29.so+0x176762])
03:56:11  _Z30protectedCompilationThreadProcP13J9PortLibraryPN2TR24CompilationInfoPerThreadE+0x82 (0x00007F89D8D76812 [libj9jit29.so+0x176812])
03:56:11  omrsig_protect+0x239 (0x00007F89DA35B3C9 [libj9prt29.so+0x2a3c9])
03:56:11  _Z21compilationThreadProcPv+0x17b (0x00007F89D8D76BDB [libj9jit29.so+0x176bdb])
03:56:11  thread_wrapper+0x163 (0x00007F89DA3233A3 [libj9thr29.so+0xb3a3])
03:56:11  start_thread+0xd9 (0x00007F89DAC5F609 [libpthread.so.0+0x8609])
03:56:11  clone+0x43 (0x00007F89DAB7E353 [libc.so.6+0x11f353])
03:56:11  ---------------------------------------
03:56:11  JVMDUMP039I Processing dump event "gpf", detail "" at 2024/05/22 00:56:04 - please wait.
03:56:11  _ZN7TR_J9VM21getBaseComponentClassEP19TR_OpaqueClassBlockRi+0x26 (0x00007F89D8E47686 [libj9jit29.so+0x247686])
03:56:11  _ZN11TR_J9VMBase17getClassSignatureEP19TR_OpaqueClassBlockP9TR_Memory+0x2a (0x00007F89D8E4948A [libj9jit29.so+0x24948a])
03:56:11  _ZL25addKnownObjectConstraintsPN3OMR16ValuePropagationEPN2TR4NodeE+0x2e4 (0x00007F89D94CF6A4 [libj9jit29.so+0x8cf6a4])
03:56:11  _Z15constrainIaloadPN3OMR16ValuePropagationEPN2TR4NodeE+0x5a6 (0x00007F89D94EC0F6 [libj9jit29.so+0x8ec0f6])
03:56:11  _ZN3OMR16ValuePropagation10launchNodeEPN2TR4NodeES3_i+0x9a (0x00007F89D952019A [libj9jit29.so+0x92019a])
03:56:11  _Z17constrainChildrenPN3OMR16ValuePropagationEPN2TR4NodeE+0x3f (0x00007F89D94F40DF [libj9jit29.so+0x8f40df])
03:56:11  _ZL15handleNullCheckPN3OMR16ValuePropagationEPN2TR4NodeEb+0xdb (0x00007F89D94CB16B [libj9jit29.so+0x8cb16b])
03:56:11  _Z16constrainNullChkPN3OMR16ValuePropagationEPN2TR4NodeE+0x17 (0x00007F89D94E7E27 [libj9jit29.so+0x8e7e27])
03:56:11  _ZN3OMR16ValuePropagation10launchNodeEPN2TR4NodeES3_i+0x9a (0x00007F89D952019A [libj9jit29.so+0x92019a])
03:56:11  _ZN3OMR16ValuePropagation12processTreesEPN2TR7TreeTopES3_+0x1a5 (0x00007F89D9520675 [libj9jit29.so+0x920675])
03:56:11  _ZN2TR21LocalValuePropagation12processBlockEPNS_7TreeTopE+0x2fe (0x00007F89D9520D6E [libj9jit29.so+0x920d6e])
03:56:11  _ZN2TR21LocalValuePropagation7performEv+0x90 (0x00007F89D9521210 [libj9jit29.so+0x921210])
03:56:11  _ZN3OMR9Optimizer19performOptimizationEPK20OptimizationStrategyiii.localalias+0x855 (0x00007F89D93ABD75 [libj9jit29.so+0x7abd75])
03:56:11  _ZN3OMR9Optimizer19performOptimizationEPK20OptimizationStrategyiii.localalias+0xc1f (0x00007F89D93AC13F [libj9jit29.so+0x7ac13f])
03:56:11  _ZN3OMR9Optimizer8optimizeEv+0x1b3 (0x00007F89D93ADA93 [libj9jit29.so+0x7ada93])
03:56:11  _ZN3OMR11Compilation7compileEv+0xa25 (0x00007F89D919F865 [libj9jit29.so+0x59f865])
03:56:11  _ZN2TR28CompilationInfoPerThreadBase7compileEP10J9VMThreadPNS_11CompilationEP17TR_ResolvedMethodR11TR_J9VMBaseP19TR_OptimizationPlanRKNS_16SegmentAllocatorE+0x4bf (0x00007F89D8D7861F [libj9jit29.so+0x17861f])
03:56:11  _ZN2TR28CompilationInfoPerThreadBase14wrappedCompileEP13J9PortLibraryPv+0x381 (0x00007F89D8D79651 [libj9jit29.so+0x179651])
03:56:11  omrsig_protect+0x239 (0x00007F89DA35B3C9 [libj9prt29.so+0x2a3c9])
03:56:11  _ZN2TR28CompilationInfoPerThreadBase7compileEP10J9VMThreadP21TR_MethodToBeCompiledRN2J917J9SegmentProviderE+0x372 (0x00007F89D8D77192 [libj9jit29.so+0x177192])
03:56:11  _ZN2TR24CompilationInfoPerThread12processEntryER21TR_MethodToBeCompiledRN2J917J9SegmentProviderE+0x128 (0x00007F89D8D774D8 [libj9jit29.so+0x1774d8])
03:56:11  _ZN2TR24CompilationInfoPerThread14processEntriesEv+0x35b (0x00007F89D8D763FB [libj9jit29.so+0x1763fb])
03:56:11  _ZN2TR24CompilationInfoPerThread3runEv+0x42 (0x00007F89D8D76762 [libj9jit29.so+0x176762])
03:56:11  _Z30protectedCompilationThreadProcP13J9PortLibraryPN2TR24CompilationInfoPerThreadE+0x82 (0x00007F89D8D76812 [libj9jit29.so+0x176812])
03:56:11  omrsig_protect+0x239 (0x00007F89DA35B3C9 [libj9prt29.so+0x2a3c9])
03:56:11  _Z21compilationThreadProcPv+0x17b (0x00007F89D8D76BDB [libj9jit29.so+0x176bdb])
03:56:11  thread_wrapper+0x163 (0x00007F89DA3233A3 [libj9thr29.so+0xb3a3])
03:56:11  start_thread+0xd9 (0x00007F89DAC5F609 [libpthread.so.0+0x8609])
03:56:11  clone+0x43 (0x00007F89DAB7E353 [libc.so.6+0x11f353])
03:56:11  ---------------------------------------
03:56:11  JVMDUMP039I Processing dump event "gpf", detail "" at 2024/05/22 00:56:04 - please wait.
03:56:11  JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/core.20240522.005601.1621137.0001.dmp
03:56:11  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/javacore.20240522.005601.1621137.0004.txt' in response to an event
03:56:11  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/core.20240522.005604.1621137.0003.dmp' in response to an event
03:56:11  JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/javacore.20240522.005601.1621137.0004.txt
03:56:11  JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/core.20240522.005604.1621137.0003.dmp
03:56:11  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/javacore.20240522.005604.1621137.0006.txt' in response to an event
03:56:11  JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/Snap.20240522.005601.1621137.0005.trc' in response to an event
03:56:11  JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/Snap.20240522.005601.1621137.0005.trc
03:56:11  JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.openjdk_x86-64_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17163641914875/jdk_lang_1/work/scratch/0/jitdump.20240522.005601.1621137.0007.dmp' in response to an event
03:56:11  JVMDUMP051I JIT dump occurred in 'MainThread' thread 0x00007F89D44B4E00
03:56:11  JVMDUMP013I Processed dump event "gpf", detail "".
```

A small internal grinder https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40750/ without JITServer appears to reproduce this issue consistently.

--- a/runtime/compiler/codegen/J9RecognizedMethodsEnum.hpp
+++ b/runtime/compiler/codegen/J9RecognizedMethodsEnum.hpp
@@ -1150,6 +1150,13 @@
    java_lang_invoke_MethodHandleImpl_profileBoolean,
    java_lang_invoke_MethodHandleImpl_isCompileConstant,
 
+   java_lang_invoke_VarHandleX_Array_method,
+   java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method,
+   java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method,
+
+   java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method,
+   java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method,
+
    // Clone and Deep Copy
    java_lang_J9VMInternals_is32Bit,
    java_lang_J9VMInternals_isClassModifierPublic,

--- a/runtime/compiler/env/j9method.cpp
+++ b/runtime/compiler/env/j9method.cpp
@@ -4523,6 +4523,245 @@ void TR_ResolvedJ9Method::construct()
             if (!strncmp(name, "invokeExact_thunkArchetype_", 27))
                setRecognizedMethodInfo(TR::java_lang_invoke_InterfaceHandle_invokeExact);
             }
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+         // The classes containing VarHandle operator or access methods also contain methods that provide other functionality.
+         // There can be unintended consequences if we were to treat them the same way as VH operator/access methods. The VarHandle
+         // operator/access methods we need to recognize all have two things in common - they are static methods and their first
+         // parameter is Ljava/lang/invoke/VarHandle;, which is different from the non VH operator methods that we should not be
+         // treating the same way. The VH operator methods without the leading VH in the signature are just helper methods that
+         // are called from other VH operator methods which fulfil the leading VH parameter check.
+         else if (sigLen > 29 && !strncmp(sig, "(Ljava/lang/invoke/VarHandle;", 29) && isStatic())
+            {
+            if ((classNameLen == 40 && !strncmp(className, "java/lang/invoke/VarHandleBooleans$Array", 40)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 56 && !strncmp(className, "java/lang/invoke/VarHandleBooleans$FieldInstanceReadOnly", 56)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 57 && !strncmp(className, "java/lang/invoke/VarHandleBooleans$FieldInstanceReadWrite", 57)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleBooleans$FieldStaticReadOnly", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 55 && !strncmp(className, "java/lang/invoke/VarHandleBooleans$FieldStaticReadWrite", 55)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 37 && !strncmp(className, "java/lang/invoke/VarHandleBytes$Array", 37)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleBytes$FieldInstanceReadOnly", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleBytes$FieldInstanceReadWrite", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 51 && !strncmp(className, "java/lang/invoke/VarHandleBytes$FieldStaticReadOnly", 51)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 52 && !strncmp(className, "java/lang/invoke/VarHandleBytes$FieldStaticReadWrite", 52)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 37 && !strncmp(className, "java/lang/invoke/VarHandleChars$Array", 37)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleChars$FieldInstanceReadOnly", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleChars$FieldInstanceReadWrite", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 51 && !strncmp(className, "java/lang/invoke/VarHandleChars$FieldStaticReadOnly", 51)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 52 && !strncmp(className, "java/lang/invoke/VarHandleChars$FieldStaticReadWrite", 52)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 39 && !strncmp(className, "java/lang/invoke/VarHandleDoubles$Array", 39)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 55 && !strncmp(className, "java/lang/invoke/VarHandleDoubles$FieldInstanceReadOnly", 55)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 56 && !strncmp(className, "java/lang/invoke/VarHandleDoubles$FieldInstanceReadWrite", 56)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleDoubles$FieldStaticReadOnly", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleDoubles$FieldStaticReadWrite", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 38 && !strncmp(className, "java/lang/invoke/VarHandleFloats$Array", 38)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleFloats$FieldInstanceReadOnly", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 55 && !strncmp(className, "java/lang/invoke/VarHandleFloats$FieldInstanceReadWrite", 55)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 52 && !strncmp(className, "java/lang/invoke/VarHandleFloats$FieldStaticReadOnly", 52)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleFloats$FieldStaticReadWrite", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 36 && !strncmp(className, "java/lang/invoke/VarHandleInts$Array", 36)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 52 && !strncmp(className, "java/lang/invoke/VarHandleInts$FieldInstanceReadOnly", 52)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleInts$FieldInstanceReadWrite", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 50 && !strncmp(className, "java/lang/invoke/VarHandleInts$FieldStaticReadOnly", 50)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 51 && !strncmp(className, "java/lang/invoke/VarHandleInts$FieldStaticReadWrite", 51)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 37 && !strncmp(className, "java/lang/invoke/VarHandleLongs$Array", 37)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleLongs$FieldInstanceReadOnly", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleLongs$FieldInstanceReadWrite", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 51 && !strncmp(className, "java/lang/invoke/VarHandleLongs$FieldStaticReadOnly", 51)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 52 && !strncmp(className, "java/lang/invoke/VarHandleLongs$FieldStaticReadWrite", 52)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 42 && !strncmp(className, "java/lang/invoke/VarHandleReferences$Array", 42)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 58 && !strncmp(className, "java/lang/invoke/VarHandleReferences$FieldInstanceReadOnly", 58)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 59 && !strncmp(className, "java/lang/invoke/VarHandleReferences$FieldInstanceReadWrite", 59)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 56 && !strncmp(className, "java/lang/invoke/VarHandleReferences$FieldStaticReadOnly", 56)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 57 && !strncmp(className, "java/lang/invoke/VarHandleReferences$FieldStaticReadWrite", 57)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 38 && !strncmp(className, "java/lang/invoke/VarHandleShorts$Array", 38)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_Array_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleShorts$FieldInstanceReadOnly", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 55 && !strncmp(className, "java/lang/invoke/VarHandleShorts$FieldInstanceReadWrite", 55)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 52 && !strncmp(className, "java/lang/invoke/VarHandleShorts$FieldStaticReadOnly", 52)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleShorts$FieldStaticReadWrite", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsChars$ArrayHandle", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method);
+               }
+            else if ((classNameLen == 59 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsChars$ByteBufferHandle", 59)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method);
+               }
+            else if ((classNameLen == 56 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsDoubles$ArrayHandle", 56)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method);
+               }
+            else if ((classNameLen == 61 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsDoubles$ByteBufferHandle", 61)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method);
+               }
+            else if ((classNameLen == 55 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsFloats$ArrayHandle", 55)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method);
+               }
+            else if ((classNameLen == 60 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsFloats$ByteBufferHandle", 60)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method);
+               }
+            else if ((classNameLen == 53 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsInts$ArrayHandle", 53)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method);
+               }
+            else if ((classNameLen == 58 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsInts$ByteBufferHandle", 58)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method);
+               }
+            else if ((classNameLen == 54 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsLongs$ArrayHandle", 54)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method);
+               }
+            else if ((classNameLen == 59 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsLongs$ByteBufferHandle", 59)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method);
+               }
+            else if ((classNameLen == 55 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsShorts$ArrayHandle", 55)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method);
+               }
+            else if ((classNameLen == 60 && !strncmp(className, "java/lang/invoke/VarHandleByteArrayAsShorts$ByteBufferHandle", 60)))
+               {
+               setRecognizedMethodInfo(TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method);
+               }
+            }
+#endif
          else if ((classNameLen >= 59 + 3 && classNameLen <= 59 + 7) && !strncmp(className, "java/lang/invoke/ArrayVarHandle$ArrayVarHandleOperations$Op", 59))
             {
             setRecognizedMethodInfo(TR::java_lang_invoke_ArrayVarHandle_ArrayVarHandleOperations_OpMethod);
@@ -5629,10 +5868,18 @@ TR_J9MethodBase::isVarHandleOperationMethod(TR::RecognizedMethod rm)
    {
    switch (rm)
       {
+#if defined (J9VM_OPT_OPENJDK_METHODHANDLE)
+      case TR::java_lang_invoke_VarHandleX_Array_method:
+      case TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method:
+      case TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method:
+      case TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method:
+      case TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method:
+#else
       case TR::java_lang_invoke_ArrayVarHandle_ArrayVarHandleOperations_OpMethod:
       case TR::java_lang_invoke_StaticFieldVarHandle_StaticFieldVarHandleOperations_OpMethod:
       case TR::java_lang_invoke_InstanceFieldVarHandle_InstanceFieldVarHandleOperations_OpMethod:
       case TR::java_lang_invoke_ByteArrayViewVarHandle_ByteArrayViewVarHandleOperations_OpMethod:
+#endif
          return true;
       default:
          return false;

--- a/runtime/compiler/optimizer/InlinerTempForJ9.cpp
+++ b/runtime/compiler/optimizer/InlinerTempForJ9.cpp
@@ -2492,11 +2492,16 @@ TR_J9InlinerPolicy::callMustBeInlined(TR_CallTarget *calltarget)
    {
    TR_ResolvedMethod *method = calltarget->_calleeMethod;
 
-   if (method->convertToMethod()->isArchetypeSpecimen())
-      return true;
-
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
    if (comp()->fej9()->isLambdaFormGeneratedMethod(method))
       return true;
+
+   if (TR_J9MethodBase::isVarHandleOperationMethod(method->getRecognizedMethod()))
+      return true;
+#else
+   if (method->convertToMethod()->isArchetypeSpecimen())
+      return true;
+#endif
 
    return callMustBeInlinedInCold(method);
    }

--- a/runtime/compiler/optimizer/UnsafeFastPath.cpp
+++ b/runtime/compiler/optimizer/UnsafeFastPath.cpp
@@ -248,7 +248,7 @@ bool TR_UnsafeFastPath::tryTransformUnsafeAtomicCallInVarHandleAccessMethod(TR::
       return false;
 
    TR::Node* unsafeAddress = NULL;
-   if (callerMethod == TR::java_lang_invoke_StaticFieldVarHandle_StaticFieldVarHandleOperations_OpMethod)
+   if (isUnsafeCallerAccessingStaticField(callerMethod))
       {
       TR::Node *jlClass = node->getChild(1);
       TR::Node *j9Class = TR::Node::createWithSymRef(node, TR::aloadi, 1, jlClass, comp()->getSymRefTab()->findOrCreateClassFromJavaLangClassSymbolRef());
@@ -772,7 +772,7 @@ int32_t TR_UnsafeFastPath::perform()
                // The offset for a static field is low taged, mask out the last bit to get the real offset
                TR::Node *newOffset =
                   TR::Node::create(offset, TR::land, 2, offset,
-                                  TR::Node::lconst(offset, ~1));
+                                  TR::Node::lconst(offset, ~J9_SUN_FIELD_OFFSET_MASK));
                node->setAndIncChild(offsetChild, newOffset);
                offset->recursivelyDecReferenceCount();
                base = ramStatics;

--- a/runtime/compiler/optimizer/UnsafeFastPath.cpp
+++ b/runtime/compiler/optimizer/UnsafeFastPath.cpp
@@ -103,6 +103,9 @@ static bool isTransformableUnsafeAtomic(TR::Compilation *comp, TR::RecognizedMet
 
 static bool isKnownUnsafeCaller(TR::RecognizedMethod rm)
    {
+#if defined (J9VM_OPT_OPENJDK_METHODHANDLE)
+   return TR_J9MethodBase::isVarHandleOperationMethod(rm);
+#else
    switch (rm)
       {
       case TR::java_lang_invoke_ArrayVarHandle_ArrayVarHandleOperations_OpMethod:
@@ -118,15 +121,40 @@ static bool isKnownUnsafeCaller(TR::RecognizedMethod rm)
          return false;
       }
    return false;
+#endif
    }
 
 static bool isUnsafeCallerAccessingStaticField(TR::RecognizedMethod rm)
    {
    switch (rm)
       {
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+      case TR::java_lang_invoke_VarHandleX_FieldStaticReadOnlyOrReadWrite_method:
+#else
       case TR::java_lang_invoke_StaticFieldVarHandle_StaticFieldVarHandleOperations_OpMethod:
       case TR::java_lang_invoke_StaticFieldGetterHandle_invokeExact:
       case TR::java_lang_invoke_StaticFieldSetterHandle_invokeExact:
+#endif
+         return true;
+      default:
+         return false;
+      }
+   return false;
+   }
+
+
+static bool isVarHandleOperationMethodOnArray(TR::RecognizedMethod rm)
+   {
+   switch (rm)
+      {
+#if defined (J9VM_OPT_OPENJDK_METHODHANDLE)
+      case TR::java_lang_invoke_VarHandleX_Array_method:
+      case TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method:
+      case TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method:
+#else
+      case TR::java_lang_invoke_ArrayVarHandle_ArrayVarHandleOperations_OpMethod:
+      case TR::java_lang_invoke_ByteArrayViewVarHandle_ByteArrayViewVarHandleOperations_OpMethod:
+#endif
          return true;
       default:
          return false;
@@ -136,6 +164,9 @@ static bool isUnsafeCallerAccessingStaticField(TR::RecognizedMethod rm)
 
 static bool isUnsafeCallerAccessingArrayElement(TR::RecognizedMethod rm)
    {
+#if defined (J9VM_OPT_OPENJDK_METHODHANDLE)
+   return isVarHandleOperationMethodOnArray(rm);
+#else
    switch (rm)
       {
       case TR::java_lang_invoke_ArrayVarHandle_ArrayVarHandleOperations_OpMethod:
@@ -144,19 +175,7 @@ static bool isUnsafeCallerAccessingArrayElement(TR::RecognizedMethod rm)
       default:
          return false;
       }
-   return false;
-   }
-
-static bool isVarHandleOperationMethodOnArray(TR::RecognizedMethod rm)
-   {
-   switch (rm)
-      {
-      case TR::java_lang_invoke_ArrayVarHandle_ArrayVarHandleOperations_OpMethod:
-      case TR::java_lang_invoke_ByteArrayViewVarHandle_ByteArrayViewVarHandleOperations_OpMethod:
-         return true;
-      default:
-         return false;
-      }
+#endif
    return false;
    }
 
@@ -164,9 +183,16 @@ static bool isVarHandleOperationMethodOnNonStaticField(TR::RecognizedMethod rm)
    {
    switch (rm)
       {
+#if defined (J9VM_OPT_OPENJDK_METHODHANDLE)
+      case TR::java_lang_invoke_VarHandleX_FieldInstanceReadOnlyOrReadWrite_method:
+      case TR::java_lang_invoke_VarHandleX_Array_method:
+      case TR::java_lang_invoke_VarHandleByteArrayAsX_ArrayHandle_method:
+      case TR::java_lang_invoke_VarHandleByteArrayAsX_ByteBufferHandle_method:
+#else
       case TR::java_lang_invoke_InstanceFieldVarHandle_InstanceFieldVarHandleOperations_OpMethod:
       case TR::java_lang_invoke_ArrayVarHandle_ArrayVarHandleOperations_OpMethod:
       case TR::java_lang_invoke_ByteArrayViewVarHandle_ByteArrayViewVarHandleOperations_OpMethod:
+#endif
          return true;
       default:
          return false;
@@ -336,6 +362,7 @@ int32_t TR_UnsafeFastPath::perform()
       {
       TR::Node *ttNode = tt->getNode();
       TR::Node *node = ttNode->getNumChildren() > 0 ? ttNode->getFirstChild() : NULL; // Get the first child of the tree
+      TR::RecognizedMethod recognizedVarHandleOpMethod = TR::unknownMethod;
       if (node && node->getOpCode().isCall() && !node->getSymbol()->castToMethodSymbol()->isHelper())
          {
          TR::SymbolReference *symRef = node->getSymbolReference();
@@ -345,6 +372,10 @@ int32_t TR_UnsafeFastPath::perform()
             {
             TR::RecognizedMethod caller = getVarHandleAccessMethodFromInlinedCallStack(comp(), node);
             TR::RecognizedMethod callee = symbol->getRecognizedMethod();
+
+            if (TR_J9MethodBase::isVarHandleOperationMethod(caller))
+               recognizedVarHandleOpMethod = caller;
+
             if (TR_J9MethodBase::isVarHandleOperationMethod(caller) &&
                 (isTransformableUnsafeAtomic(comp(), callee) ||
                  symbol->getMethod()->isUnsafeCAS(comp())))
@@ -696,6 +727,8 @@ int32_t TR_UnsafeFastPath::perform()
          // Handle VarHandle operation methods
          bool isStatic = false;
          TR::RecognizedMethod callerMethod = methodSymbol->getRecognizedMethod();
+         if (recognizedVarHandleOpMethod != TR::unknownMethod)
+            callerMethod = recognizedVarHandleOpMethod;
          TR::RecognizedMethod calleeMethod = symbol->getRecognizedMethod();
          if (isKnownUnsafeCaller(callerMethod) &&
              TR_J9MethodBase::isUnsafeGetPutWithObjectArg(calleeMethod))

